rest/rest-sqs-testing-amazon-java-sdk/src/test/scala/org/elasticmq/rest/sqs/MessageMoveTaskTest.scala

```diff
@@ -0,0 +1,208 @@
+package org.elasticmq.rest.sqs
+
+import org.elasticmq.rest.sqs.client._
+import org.elasticmq.rest.sqs.model.RedrivePolicy
+import org.elasticmq.rest.sqs.model.RedrivePolicyJson.format
+import org.scalatest.concurrent.Eventually
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import spray.json.enrichAny
+
+import java.util.UUID
+import scala.concurrent.duration.DurationInt
+
+abstract class MessageMoveTaskTest
+    extends AnyFunSuite
+    with HasSqsTestClient
+    with AwsConfig
+    with Matchers
+    with Eventually {
+
+  private val NumMessages = 6
+  private val DlqArn = s"arn:aws:sqs:$awsRegion:$awsAccountId:testQueue-dlq"
+
+  test("should run message move task") {
+    // given
+    val (queue, dlq) = createQueuesAndPopulateDlq()
+
+    // when: start message move task
+    testClient.startMessageMoveTask(DlqArn)
+
+    // then: ensure that messages are moved back to the original queue
+    eventually(timeout(5.seconds), interval(100.millis)) {
+      fetchApproximateNumberOfMessages(queue) shouldEqual NumMessages
+      fetchApproximateNumberOfMessages(dlq) shouldEqual 0
+    }
+  }
+
+  test("should run message move task with max number of messages per second") {
+    // given
+    val (queue, dlq) = createQueuesAndPopulateDlq()
+
+    // when: start message move task
+    testClient.startMessageMoveTask(DlqArn, maxNumberOfMessagesPerSecond = Some(1))
+
+    // then: ensure that not all messages were moved back to the original queue after 2 seconds
+    Thread.sleep(2000)
+    fetchApproximateNumberOfMessages(queue) should (be > 1 and be < 6)
```

Comment:
Shouldn't that be max 2 and the former one min 4?

Reply:
Only if the message move jobs starts immediately, which I'm not sure is guaranteed in the environments like the Github Actions provide (fractions of CPU core)